src/EventSubscriber/ApiKeyVersionSubscriber.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+
+class ApiKeyVersionSubscriber implements EventSubscriberInterface
+{
+    private string $apiVersionKey;
+
+    public function __construct(string $apiVersionKey)
+    {
+        $this->apiVersionKey = $apiVersionKey;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ResponseEvent::class => 'addApiVersionKey'
+        ];
+    }
+
+    public function addApiVersionKey(ResponseEvent $event): void
+    {
+        $event->getResponse()->headers->set('X-API-VERSION', $this->apiVersionKey);
```

Comment:
This adds the header for all responses. The requirement is to add it ONLY for the response of the API endpoints (not admin, forget password, etc.).

Reply:
ok